src/components/permissions/rowLevelPolicyHelpers.ts

```diff
@@ -119,11 +119,55 @@ const generateRowLevelPolicies = (permissionRows: permissionPolicyColumns[]) =>
   return policies
 }
 
+export const modifyValueInObject = (
+  obj: object,
+  matchFn: (key: string, value: object) => boolean,
+  modifyFn: (value: object) => string
+): object => {
+  if (typeof obj != 'object') {
+    return obj
+  }
+  if (Array.isArray(obj)) {
+    console.warn('Arrays are not supported')
+    return obj
+  }
```

Comment:
We have an `isObject` function in `utilityFunctions.ts` that can deal with this:
```ts
// Returns true if input is a "proper" object (i.e. not an array or null)
export const isObject = (element: unknown): element is Object =>
  typeof element === 'object' && !Array.isArray(element) && element !== null
```

In fact, can you put this whole function in `utilityFunctions` and add a comment explaining what it does -- seems generic enough to keep in a common resource.

Reply:
🙏 5c653792f190c5a0b3ef076be653d9606b97c003, so much more confident about refactor with isObject when there are tests for modifyValueInObject